src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -131,26 +145,46 @@ def generate_json_report(self):
             **self.results_handler.all_raw_results
         }
 
-    def _populate_core_stats_and_sim_state(self, area):
+    def _read_settlement_markets_stats_to_dict(self, area: "Area") -> Dict[str, Dict]:
+        """Loop over all settlement markets and return market_stats for each in a dict."""
+        stats_dict = {}
+        for time_slot, market in area.settlement_markets.items():
+            stats_dict[time_slot.format(DATE_TIME_FORMAT)] = (
+                self._read_market_stats_to_dict(market))
+        return stats_dict
+
+    @staticmethod
+    def _read_market_stats_to_dict(market: "Market") -> Dict:
```

Comment:
I have a question: for spot markets, we only use `_read_market_stats_to_dict` to get the stats of one single market (`area.current_market`). For settlement markets, instead, we cycle through all the `area.settlement_markets`.
Could you please clarify why this is like this? What about the other past spot markets?

Reply:
IMO, settlement market should too send the results  of `last_past_settlement_market`.

Tagging @spyrostz in case something related was discussed already that I am not aware of

Reply:
I think that last_past_settlement_market should be more appropriate here. The only reason I can think of to send the open settlement markets to the results service is to report the progress of the settlement market gradually, but I do not think that this is the intent here.